src/Socket/AbstractSocket.php

```diff
@@ -251,15 +251,10 @@ public function receive(int $length = self::DEFAULT_RECEIVE_LENGTH): string
                     $makeBlockingAfterRead = false;
                 }
 
-                // fread FALSE means socket has been closed
```

Comment:
cc @enricodias. Seems this is not the case on PHP 8.1. The tests seem to pass if we don't try and close the connection on false return here. What do you think about us making this change, for 1.2.0?